packages/tests/server/dto.test.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { routerToServerAndClientNew } from './___testHelpers';
+import { initTRPC } from '@trpc/server';
+import { expectTypeOf } from 'expect-type';
+import { konn } from 'konn';
+
+describe('no transformer specified', () => {
+  const ctx = konn()
+    .beforeEach(() => {
+      const t = initTRPC.create();
+
+      const appRouter = t.router({
+        happy: t.procedure.query(() => ({
+          map: new Map<string, string>([['foo', 'bar']]),
+          set: new Set<string>(['foo', 'bar']),
+          date: new Date(0),
+          fn: () => {
+            return 'hello';
+          },
+          arrayWithUndefined: [1, undefined, 2],
```

Comment:
This should be DTO'd to `number[]`

Reply:
A tuple, `[1, undefined, 2] as const`, on the other hand should be DTO'd to `[1, 2]`